yt/frontends/gadget/data_structures.py

```diff
@@ -168,14 +169,40 @@ def validate(self):
         return True
 
 
-class GadgetBinaryFile(ParticleFile):
+class GadgetParticleFile(ParticleFile):
+    # an intermediate abstract class for both gadget types.
+
+    _file_read_mode = None  # either "r" or "rb", set by child class
```

Comment:
I'm actually not sure that we ever need it to be `r` as opposed to `rb`.  I think the distinction may not be as important with Python 3 in our particular case.  The data will always be binary.

Reply:
So conceptually I agree... but `h5py.File` only accepts `r` (https://docs.h5py.org/en/stable/high/file.html). Maybe this attribute would be better as a tuple of arguments to emphasize that these arguments are tied to the function handle set by `_opening_func`? e.g., 

```
_file_opening_kwargs = ( ("mode", "r"), ) 
```

Reply:
Oh!  Hmmmm.  Nah, don't change it for me.  But ... I think it might be OK to just use `r` for `open` and numpy should still be able to use `fromfile`.

Reply:
oooh, "r" for open, not "rb" for h5py. I'll try it out :)

Reply:
ya, that works! Will push changes in a moment!